nistats/permutation_tests.py

```diff
@@ -0,0 +1,296 @@
+        nifti file contains the uncorrected p values derived from the
+        permutations at the voxel level, while the second nifti file contains
+        the corrected p values derived from the permutations whole brain max
+        statistic.
+    """
+    # check n_jobs (number of CPUs)
+    if n_jobs == 0:  # invalid according to joblib's conventions
+        raise ValueError("'n_jobs == 0' is not a valid choice. "
+                         "Please provide a positive number of CPUs, or -1 "
+                         "for all CPUs, or a negative number (-i) for "
+                         "'all but (i-1)' CPUs (joblib conventions).")
+    elif n_jobs < 0:
+        n_jobs = max(1, joblib.cpu_count() - int(n_jobs) + 1)
+    else:
+        n_jobs = min(n_jobs, joblib.cpu_count())
+
+    # Check n_perm
+    if n_perm <= 0:
+        raise ValueError("'n_perm <= 0' is not a valid choice.")
+
+    # Process one contrast at a time
+    contrast_results = {}
+    for (con_id, con_val) in contrasts.iteritems():
+        # Check permutation can be performed on contrasts
+        if not _meet_contrast_conditions(con_val):
+            warnings.warn('Contrast %s do not meet conditions.'
+                          'Computation ignored.' % (con_id,))
+            continue
+
+        # Get original stat
+        design_matrices = []
+        for time_frames, paradigm, kwargs in design_matrix_objs:
+            design_matrices.append(make_design_matrix(time_frames, paradigm,
+                                                      **kwargs))
+        glm_obj = clone(glm_ref)
+        original_stat, = glm_obj.fit(imgs, design_matrices) \
+                                .transform(con_val, contrast_name=con_id,
+                                           output_z=False, output_stat=True)
+        original_stat = glm_obj.masker_.transform(original_stat)[0]
+        if two_sided_test:
+            original_stat = np.fabs(original_stat)
+        # Check possible permutations
+
+        # Distribute permutations across jobs
+        if n_perm > n_jobs:
+            n_perm_chunks = np.asarray([n_perm / n_jobs] * n_jobs,
+                                       dtype=int)
+            n_perm_chunks[-1] += n_perm % n_jobs
+        elif n_perm > 0:
+            warnings.warn('The specified number of permutations is %d and '
+                          'the number of jobs to be performed in parallel was '
+                          'set to %s. This is incompatible so only %d jobs will '
+                          'be running. You may want to perform more permutations '
+                          'in order to take the most of the available computing '
+                          'resources.' % (n_perm, n_jobs, n_perm))
+            n_perm_chunks = np.ones(n_perm, dtype=int)
+
+        # initialize the seed of the random generator
+        rng = check_random_state(random_state)
+
+        per = joblib.Parallel(n_jobs=n_jobs, verbose=verbose)(
```

Comment:
`permutation_results` rather than `per`
The names should always be as explicit as possible.